### PR TITLE
15,18, and 24 word seed support for bip39.

### PR DESCRIPTION
**Pull request by fyookball**

### CODE REVIEW

File: electroncash/base_wizard.py [hunk 1]
```diff
@@ -512,7 +512,15 @@ def create_seed(self, seed_type):
...
         elif seed_type == 'bip39':
-            seed = mnemonic.Mnemonic('en').make_seed()
+            bip39_num_bits = 128
+            bip39_seed_length = self.config.get("bip39_seed_length")
+            if (bip39_seed_length == 15):
+                bip39_num_bits = 160
+            elif (bip39_seed_length == 18):
+                bip39_num_bits = 192
+            elif (bip39_seed_length == 24):
+                bip39_num_bits = 256
```
Comment (cculianu) [inline electroncash/base_wizard.py:522]:
```suggestion
            if bip39_seed_length == 15:
                bip39_num_bits = 160
            elif bip39_seed_length == 18:
                bip39_num_bits = 192
            elif bip39_seed_length == 24:
                bip39_num_bits = 256
```

File: electroncash_gui/qt/main_window.py [hunk 2]
```diff
@@ -4527,7 +4527,47 @@ def on_unit(x, nz):
...
+        
+        # Bip-39 Number of Seed Words
+        
+        def on_bip39_seed_length_change(x): 
+            # Get the dropdown selection
+            if (x==1):
+                bip39_val_to_config = 15
+            elif (x==2):
+                bip39_val_to_config = 18
+            elif (x==3):
```
Comment (cculianu) [inline electroncash_gui/qt/main_window.py:4539]:
```suggestion
            if x == 1:
                bip39_val_to_config = 15
            elif x == 2:
                bip39_val_to_config = 18
            elif x == 3:
```

File: electroncash_gui/qt/main_window.py [hunk 3]
```diff
@@ -4527,7 +4527,47 @@ def on_unit(x, nz):
...
+        if bip39_val_from_config not in [12,15,18,24]: 
+            bip39_val_from_config = 12
+          
+        # Set the bip39 seed length dropdown selection
+        if (bip39_val_from_config == 15):
+            bip39_combo_index = 1
+        elif (bip39_val_from_config == 18):
+            bip39_combo_index = 2
+        elif (bip39_val_from_config == 24):
+            bip39_combo_index = 3
```
Comment (cculianu) [inline electroncash_gui/qt/main_window.py:4561]: Formatting..

```suggestion
        if bip39_val_from_config not in [12, 15, 18, 24]: 
            bip39_val_from_config = 12
          
        # Set the bip39 seed length dropdown selection
        if bip39_val_from_config == 15:
            bip39_combo_index = 1
        elif bip39_val_from_config == 18:
            bip39_combo_index = 2
        elif bip39_val_from_config == 24:
            bip39_combo_index = 3
```

File: electroncash_gui/qt/main_window.py [hunk 4]
```diff
@@ -4527,7 +4527,47 @@ def on_unit(x, nz):
...
+        else: # 12 words or catch-all
+            bip39_combo_index = 0
+         
+        # Attach the Bip39 seed length to the widget.
+        bip39_combo.setCurrentIndex(bip39_combo_index)
+        bip39_combo.currentIndexChanged.connect(lambda x: on_bip39_seed_length_change(x))
```
Comment (cculianu) [inline electroncash_gui/qt/main_window.py:4567]: You don't need the lambda x thing..  can just connect it to the function directly...

```suggestion
        bip39_combo.currentIndexChanged.connect(on_bip39_seed_length_change)
```

**Comment by cculianu** [comment]
Question: Does legacy EC before this codechange implicitly support restoring from 24 seed words?  And I presume you tested this by generating a wallet with various-length seed words and then restoring from it, yes?

**Comment by fyookball** [comment]
> Question: Does legacy EC before this codechange implicitly support restoring from 24 seed words? And I presume you tested this by generating a wallet with various-length seed words and then restoring from it, yes?

yes and yes

**Comment by cculianu** [comment]
Please accept my changes so I may proceed.

**Comment by cculianu** [comment]
This doesn't belong in wallet config. This should be a GUI option in the base_wizard / install wizard. When the user is creating a new wallet is when they need to make this decision.

NACK from me on this until the above is addressed.

**Pull request merged**